src/Files.App/Data/Factories/ContentPageContextFlyoutFactory.cs

```diff
@@ -398,114 +285,113 @@ private static bool Check(ContextMenuFlyoutItemViewModel item, CurrentInstanceVi
+					ShowItem = !currentInstanceViewModel.IsPageTypeRecycleBin
 								&& !currentInstanceViewModel.IsPageTypeZipFolder
 								&& (selectedItemsPropertiesViewModel?.IsSelectedItemImage ?? false)
-				}.Build(),
-				new ContextMenuFlyoutItemViewModelBuilder(Commands.RunAsAdmin).Build(),
-				new ContextMenuFlyoutItemViewModelBuilder(Commands.RunAsAnotherUser).Build(),
-				new ContextMenuFlyoutItemViewModel()
+				},
+				new(ContextFlyoutItemType.Button, Commands.RunAsAdmin),
+				new(ContextFlyoutItemType.Button, Commands.RunAsAnotherUser),
+				new(ContextFlyoutItemType.Separator)
 				{
-					ItemType = ContextMenuFlyoutItemType.Separator,
 					ShowInSearchPage = true,
 					ShowInFtpPage = true,
 					ShowInZipPage = true,
 					ShowItem = itemsSelected
 				},
-				new ContextMenuFlyoutItemViewModelBuilder(Commands.CutItem)
+				new(ContextFlyoutItemType.Button, Commands.CutItem)
 				{
 					IsPrimary = true,
-				}.Build(),
-				new ContextMenuFlyoutItemViewModelBuilder(Commands.CopyItem)
+				},
+				new(ContextFlyoutItemType.Button, Commands.CopyItem)
 				{
 					IsPrimary = true,
-				}.Build(),
-				new ContextMenuFlyoutItemViewModelBuilder(Commands.PasteItemToSelection)
+				},
+				new(ContextFlyoutItemType.Button, Commands.PasteItemToSelection)
 				{
 					IsPrimary = true,
-					IsVisible = true,
-				}.Build(),
-				new ContextMenuFlyoutItemViewModelBuilder(Commands.CopyPath)
+					ShowItem = true,
+				},
+				new(ContextFlyoutItemType.Button, Commands.CopyPath)
 				{
-					IsVisible = UserSettingsService.GeneralSettingsService.ShowCopyPath
+					ShowItem = UserSettingsService.GeneralSettingsService.ShowCopyPath
 						&& itemsSelected
 						&&!currentInstanceViewModel.IsPageTypeRecycleBin,
-				}.Build(),
-				new ContextMenuFlyoutItemViewModelBuilder(Commands.CreateFolderWithSelection)
+				},
+				new(ContextFlyoutItemType.Button, Commands.CreateFolderWithSelection)
 				{
-					IsVisible = UserSettingsService.GeneralSettingsService.ShowCreateFolderWithSelection && itemsSelected
-				}.Build(),
-				new ContextMenuFlyoutItemViewModelBuilder(Commands.CreateShortcut)
+					ShowItem = UserSettingsService.GeneralSettingsService.ShowCreateFolderWithSelection && itemsSelected
+				},
+				new(ContextFlyoutItemType.Button, Commands.CreateShortcut)
 				{
-					IsVisible = UserSettingsService.GeneralSettingsService.ShowCreateShortcut
+					ShowItem = UserSettingsService.GeneralSettingsService.ShowCreateShortcut
 						&& itemsSelected
 						&& (!selectedItems.FirstOrDefault()?.IsShortcut ?? false)
 						&& !currentInstanceViewModel.IsPageTypeRecycleBin,
-				}.Build(),
-				new ContextMenuFlyoutItemViewModelBuilder(Commands.Rename)
+				},
+				new(ContextFlyoutItemType.Button, Commands.Rename)
 				{
 					IsPrimary = true,
-					IsVisible = itemsSelected
-				}.Build(),
-				new ContextMenuFlyoutItemViewModelBuilder(Commands.ShareItem)
+					ShowItem = itemsSelected
```

Comment:
Are you sure switching from IsVisible to ShowItem is correct? I'm not familiar enough with this property to know what it's used for.

Reply:
Yes, the same.